howtos/pubsub/how_to_consume_persistent_message_with_negative_acknowledgement.py

```diff
@@ -0,0 +1,108 @@
+    """class contains methods to settle message with negative acknowledgement"""
+
+    @staticmethod
+    def settle_message_sync(service: MessagingService, publisher, message, required_outcomes, outcome_to_settle):
+        try:
+            queue_name = constants.QUEUE_NAME_FORMAT.substitute(iteration=topic_name)
+            HowToConsumeMessageExclusiveVsSharedMode.create_queue_and_add_topic(queue_name)
+            queue_to_consume = Queue.durable_exclusive_queue(queue_name)
+
+            receiver: PersistentMessageReceiver = service.create_persistent_message_receiver_builder() \
+                .with_required_message_outcome_support(*required_outcomes).build(queue_to_consume)
+            receiver.start()
+            print(f'PERSISTENT receiver started for sync receiver... Listening to Queue [{queue_to_consume.get_name()}]')
+            receiver.add_subscription(TopicSubscription.of(topic_name))
+
+            HowToPublishPersistentMessage.publish_string_message_non_blocking(publisher, topic, message)
+
+            message: InboundMessage = receiver.receive_message()
+            print(f"the message payload is {message.get_payload_as_string()}")
+            receiver.settle(message, outcome_to_settle)
+        finally:
+            receiver.terminate()
+            HowToConsumeMessageExclusiveVsSharedMode.delete_queue(queue_to_consume.get_name())
+
+    @staticmethod
+    def settle_message_async(service: MessagingService, publisher, message, required_outcomes, outcome_to_settle):
+        event = threading.Event()
+        receiver = None
+
+        def receiver_callback(self, message: InboundMessage):
+            # Fail messages will be redelivered so we just want to settle the actual message that is not redelivered
+            if not message.is_redelivered():
+                print(f"the message payload is {message.get_payload_as_string()}")
+                receiver.settle(message, outcome_to_settle)
+                event.set()
+
+        try:
+            queue_name = constants.QUEUE_NAME_FORMAT.substitute(iteration=topic_name)
+            HowToConsumeMessageExclusiveVsSharedMode.create_queue_and_add_topic(queue_name)
+            queue_to_consume = Queue.durable_exclusive_queue(queue_name)
+
+            receiver: PersistentMessageReceiver = service.create_persistent_message_receiver_builder() \
+                .with_required_message_outcome_support(*required_outcomes).build(queue_to_consume)
+            receiver.start()
+            print(f'PERSISTENT receiver started for async receiver... Listening to Queue [{queue_to_consume.get_name()}]')
+            receiver.add_subscription(TopicSubscription.of(topic_name))
+            message_handler = BasicTestMessageHandler(test_callback=receiver_callback)
+            receiver.receive_async(message_handler)
+
+            HowToPublishPersistentMessage.publish_string_message_non_blocking(publisher, topic, message)
+            event.wait(5)
```

Comment:
Maybe print the stats at the end.

Reply:
Ah ok sounds good, I will print the stats down below